src/main/cpp/src/xxhash64.cu

```diff
@@ -0,0 +1,359 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "hash.cuh"
+
+#include <cudf/column/column_factories.hpp>
+#include <cudf/detail/hashing.hpp>
+#include <cudf/detail/utilities/algorithm.cuh>
+#include <cudf/detail/utilities/hash_functions.cuh>
+#include <cudf/table/table_device_view.cuh>
+
+#include <rmm/cuda_stream_view.hpp>
+#include <rmm/exec_policy.hpp>
+
+#include <thrust/tabulate.h>
+
+namespace spark_rapids_jni {
+
+namespace {
+
+using hash_value_type = int64_t;
+using half_size_type  = int32_t;
+
+constexpr __device__ inline int64_t rotate_bits_left_signed(hash_value_type h, int8_t r)
+{
+  return (h << r) | (h >> (64 - r)) & ~(-1 << r);
+}
+
+template <typename Key>
+struct XXHash_64 {
+  using result_type = hash_value_type;
+
+  constexpr XXHash_64() = delete;
+  constexpr XXHash_64(hash_value_type seed) : m_seed(seed) {}
+
+  template <typename T>
+  __device__ inline T getblock32(std::byte const* data, cudf::size_type offset) const
+  {
+    // Read a 4-byte value from the data pointer as individual bytes for safe
+    // unaligned access (very likely for string types).
+    auto block      = reinterpret_cast<std::uint8_t const*>(data + offset);
+    uint32_t result = static_cast<uint32_t>(block[0]) | (static_cast<uint32_t>(block[1]) << 8) |
```

Comment:
Is there a functional difference that I'm missing between this and https://github.com/NVIDIA/spark-rapids-jni/pull/1246/files#diff-3b5fbfa28ce0798e562307d27b7636b7fd46e6d90d92e93c61a0da5411aec0dbR55. The addition of the extra static I believe is unnecessary.